packages/core/contracts/financial-templates/common/FundingRateApplier.sol

```diff
@@ -217,18 +217,17 @@ abstract contract FundingRateApplier is FeePayer {
         uint256 proposalTime = fundingRate.proposalTime;
         if (proposalTime != 0) {
             // Attempt to update the funding rate.
-            OptimisticOracle optimisticOracle = _getOptimisticOracle();
+            OptimisticOracleInterface optimisticOracle = _getOptimisticOracle();
             bytes32 identifier = fundingRate.identifier;
             bytes memory ancillaryData = _getAncillaryData();
 
             // Try to get the price from the optimistic oracle.
+            OptimisticOracleInterface.Request memory request =
+                optimisticOracle.getRequest(address(this), identifier, proposalTime, ancillaryData);
             try optimisticOracle.getPrice(identifier, proposalTime, ancillaryData) returns (int256 price) {
```

Comment:
its worth having a comment here i think, the logic in getPrice is really complicated to trace. Basically it will throw if there is no price request, or the price request response is being disputed. This causes an error in settlement as its unable to resolve the proposal and throws.

Reply:
[Comments in this commit](https://github.com/UMAprotocol/protocol/pull/2280/commits/6d62763a2773e9ef5e62c3481ff530d9fa5867f3)! Thanks @daywiss feedback about the readability is really useful.